hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -80,6 +80,9 @@ public void pickAddress() throws Exception {
                 logger.finest("Using public address the same as the bind address: " + publicAddress);
             }
         } catch (Exception e) {
+            if (serverSocketChannel != null) {
+                serverSocketChannel.close();
```

Comment:
This method is already marked as "throws Exception" (probably bit overkill, but anyway..), and the exception is caught and rethrown somewhere else.